Messaging.md

```diff
@@ -44,20 +55,114 @@ The following convenience types are also defined:
+
+#### Version 0 `oracle_info`
+
+1. type: 42770 (`oracle_info_v0`)
+2. data:
+   * [`x_point`:`oracle_public_key`]
+   * [`x_point`:`oracle_nonce`]
+
+This type of oracle info is for single-oracle, single signature (and hence single nonce) events.
+
+### The `funding_input` Type
+
+This type contains information about a specific input to be used in a funding transaction, as well as its corresponding on-chain UTXO.
+
+#### Version 0  `funding_input`
+
+1. type: 42772 (`funding_input_v0`)
+2. data:
+   * [`u16`:`prevtx_len`]
+   * [`prevtx_len*byte`:`prevtx`]
+   * [`u32`:`prevtx_vout`]
+   * [`u32`:`sequence`]
+   * [`u16`:`max_witness_len`]
+   * [`u16`:`redeemscript_len`]
+   * [`redeemscript_len*byte`:`script`]
+
+`prevtx_tx` is the serialized transaction whose `prevtx_vout` output is being spent.
+The transaction is used to validate this output's value and to validate that it is a SegWit output.
+
+`max_witness_len` is the total serialized length of the witness data that will be supplied
+(e.g. sizeof(varint) + sizeof(witness) for each) in `funding_signatures`.
+
+`script` is the script signature field for the input. Only applicable for P2SH-wrapped inputs.
+The length of the script should not be included in the transmitted script data.
+
+### The `cet_adaptor_signatures` Type
+
+This type contains CET signatures and any necessary information linking the signatures to their corresponding outcome.
+
+#### Version 0 `cet_adaptor_signatures`
+
+1. type: 42774 (`cet_adaptor_signatures_v0`)
+2. data:
+   * [`ecdsa_adaptor_signature`:`signature_1`]
+   * [`dleq_proof`:`dleq_prf_1`]
+   * ...
+   * [`ecdsa_adaptor_signature`:`signature_n`]
+   * [`dleq_proof`:`dleq_prf_n`]
+
+This type should be used with [`contract_info_v0`](#version-0-contract_info) where each indexed signature in the data corresponds to the outcome of the same index. As in [`contract_info_v0`](#version-0-contract_info), the number of signatures is omitted as it can be derived from the length field of the TLV.
+
+### The `funding_signatures` Type
+
+This type contains signatures of the funding transaction and any necessary information linking the signatures to their inputs.
+
+#### Version 0 `funding_signatures`
+
+1. type: 42776 (`funding_signatures_v0`)
+2. data:
+   * [`u16`:`num_witnesses`]
+   * [`u16`:`num_witness_elems_1`]
+   * [`num_witness_elems_1*witness_element`:`witness_elements_1`]
+   * ...
+   * [`u16`:`num_witness_elems_num_witnesses`]
+   * [`num_witness_elems_num_witnesses*witness_element`:`witness_elements_num_witnesses`]
+3. subtype: `witness_element`
+4. data:
+   * [`u16`:`len`]
+   * [`len*byte`:`witness`]
+
+`witness` is the data for a witness element in a witness stack. An empty `witness_stack` is an error,
+as every input must be Segwit. Witness elements should *not* include their length as part of the witness data.
```

Comment:
Hmmm, I'm not sure this is a feasible constraint seeing as one party never sees the other's witnesses before they see them on-chain. Is there any reason witness malleability is an issue in this context as it doesn't invalidate future txs?

Reply:
Yes witness malleability is a pinning vector and as such you can prevent confirmation of the funding transaction and escape the contract in case of non-favorable outcome. 

In theory you can parse a `witnessScript` and decide if a solving witness is malleable. But as pointed IIRC  Miniscript isn't supporting this.